lib/pages/portal/service/website_provider.dart

```diff
@@ -103,6 +104,52 @@ class WebsiteProvider with ChangeNotifier {
     }
   }
 
+
+  /// Initializes the number of visits of websites with the value stored locally.
+  /// Because [PrefService] doesn't support storing maps, the
+  /// data is stored in 2 lists: the list of website IDs ([websiteIds]) and the list
+  /// with the number of visits ([websiteVisits]), where `websiteVisits[i]` is the
+  /// number of times the user accessed website with ID `websiteIds[i]`.
+  void initializeNumberOfVisits(List<Website> websites) {
+    List<String> websiteIds =
+        PrefService.sharedPreferences.getStringList("websiteIds");
+    List<String>  websiteVisits =
+        PrefService.sharedPreferences.getStringList("websiteVisits");
+
+    if (websiteIds != null &&  websiteVisits != null) {
+      var visitsByWebsiteId = Map<String, int>.from(websiteIds.asMap().map((index, key) =>
+          MapEntry(key, int.tryParse( websiteVisits[index] ?? 0))));
+      websites.forEach((website) {
+        if (visitsByWebsiteId[website.id] != null) {
+          website.numberOfVisits = visitsByWebsiteId[website.id];
+        }
+      });
+    }
+  }
+
+  /// Increments the number of visits of [website], both in-memory and on the local storage.
+  /// Because [PrefService] doesn't support storing maps, the
+  /// data is stored in 2 lists: the list of website IDs ([websiteIds]) and the list
+  /// with the number of visits ([websiteVisits]), where `websiteVisits[i]` is the
+  /// number of times the user accessed website with ID `websiteIds[i]`.
```

Comment:
```suggestion
  /// Increments the number of visits of [website], both in-memory and on the local storage.
  ///
  /// Because [PrefService] doesn't support storing maps, the
  /// data is stored in 2 lists: the list of website IDs ([websiteIds]) and the list
  /// with the number of visits ([websiteVisits]), where `websiteVisits[i]` is the
  /// number of times the user accessed website with ID `websiteIds[i]`.
```